### PR TITLE
COG-1221 Fix emoji's character count on Notes Field.

### PR DESCRIPTION
**Pull request by RamuKammagani-osi**
#### Which User story does this relate to (link)?
- https://osi-cwds.atlassian.net/browse/COG-1221

#### Brief feature description
- Displays same count of emoji's in all 5 browsers, Each emojis is counted as 2 characters.

### CODE REVIEW

**Comment by kelfink** [comment]
normalize().  nice method!

File: app/javascript/selectors/detailSelector.test.js [hunk 1]
```diff
@@ -958,6 +958,16 @@ describe('selectors', () => {
       expect(userNotes(state)).toEqual({ notes: '23', notesLength: 2 })
     })
 
+    it('returns notes and length if notes is emojis', () => {
+      const state = getState({ notes: '😁' })
```
Comment (kelfink) [inline app/javascript/selectors/detailSelector.test.js]: I am jealous you have a LEGITIMATE reason to put emoji's in code

File: app/javascript/selectors/detailSelector.test.js [hunk 2]
```diff
@@ -958,6 +958,16 @@ describe('selectors', () => {
...
+      expect(userNotes(state)).toEqual({ notes: '😁', notesLength: 2 })
```
Comment (kelfink) [inline app/javascript/selectors/detailSelector.test.js]: How is this story changing things?  Not sure I understand.  '😁'.length is 2, and '😁'.normalize().length is 2

File: app/javascript/selectors/detailSelector.js [hunk 3]
```diff
@@ -189,5 +189,5 @@ export const phoneExtension = state => {
 
 export const userNotes = state => {
   const notes = safeGet(state, 'fetchDetails.details.records.user.notes') || ''
-  return { notes: notes, notesLength: notes.length }
+  return { notes: notes, notesLength: notes.normalize().length }
```
Comment (kelfink) [inline app/javascript/selectors/detailSelector.js]: normalize seems not to do anything for us.  See if you get the same functionality if you get rid of it everywhere.
```suggestion
  return { notes: notes, notesLength: notes.length }
```

File: app/javascript/views/userDetail/Notes.jsx [hunk 4]
```diff
@@ -2,6 +2,15 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import { Rolodex, Card, CardBody, CardHeader, CardTitle, Input } from '@cwds/components'
 
+export const allowUserInput = event => {
+  const notes = event.target.value
+  const value = notes.normalize()
```
Comment (kelfink) [inline app/javascript/views/userDetail/Notes.jsx]: remove line

File: app/javascript/views/userDetail/Notes.jsx [hunk 5]
```diff
@@ -2,6 +2,15 @@ import React from 'react'
...
+  if (value.length > 250) {
+    return value.substring(0, 250)
+  }
+  return value
```
Comment (kelfink) [inline app/javascript/views/userDetail/Notes.jsx]:
```suggestion
  return event.target.value.substring(0,250)
```

File: app/javascript/views/userDetail/Notes.jsx [hunk 6]
```diff
@@ -2,6 +2,15 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import { Rolodex, Card, CardBody, CardHeader, CardTitle, Input } from '@cwds/components'
 
+export const allowUserInput = event => {
```
Comment (kelfink) [inline app/javascript/views/userDetail/Notes.jsx]: not a very descriptive name, is it?  Maybe truncateInputLength.  and down the road we could add a parameter to set the max length

**Comment by RamuKammagani-osi** [comment]
@kelfink All the requested changes have been updated in the latest commit

**Pull request merged**